include/zephyr/sys/arch_interface.h

```diff
@@ -294,20 +295,26 @@ static inline bool arch_irq_unlocked(unsigned int key);
  */
 void arch_irq_disable(unsigned int irq);
 
+void arch_irq_intc_disable(const struct device *intc, unsigned int irq);
+
 /**
  * Enable the specified interrupt line
  *
  * @see irq_enable()
  */
 void arch_irq_enable(unsigned int irq);
 
+void arch_irq_intc_enable(const struct device *intc, unsigned int irq);
```

Comment:
the changes here look OK to me too, although I would personally put the `intc` argument at the end (to avoid registers being shuffled between calls.